wherehows-data-model/DDL/ETL_DDL/metric_metadata.sql

```diff
@@ -34,18 +34,52 @@ CREATE TABLE dict_business_metric  (
   `metric_display_factor_sym`	VARCHAR(20) COMMENT '%, (K), (M), (B), (GB), (TB), (PB)' NULL,
   `metric_good_direction`	VARCHAR(20) COMMENT 'UP, DOWN, ZERO, FLAT' NULL,
   `metric_formula`           	TEXT COMMENT 'Expression, Code Snippet or Calculation Logic' NULL,
-  `dimensions`			VARCHAR(500) CHAR SET latin1 NULL,
+  `dimensions`			VARCHAR(800) CHAR SET latin1 NULL,
   `owners`                 	VARCHAR(300) NULL,
   `tags`			VARCHAR(300) NULL,
   `urn`                         VARCHAR(300) CHAR SET latin1 NOT NULL,
   `metric_url`			VARCHAR(300) CHAR SET latin1 NULL,
   `wiki_url`              	VARCHAR(300) CHAR SET latin1 NULL,
   `scm_url`               	VARCHAR(300) CHAR SET latin1 NULL,
+  `wh_etl_exec_id`              BIGINT COMMENT 'wherehows etl execution id that modified this record',
   PRIMARY KEY(metric_id),
+  UNIQUE KEY `uq_dataset_urn` (`urn`),
   KEY `idx_dict_business_metric__urn` (`urn`) USING BTREE,
   KEY `idx_dict_business_metric__ref_id` (`metric_ref_id`) USING BTREE,
   FULLTEXT KEY `fti_dict_business_metric_all` (`metric_name`, `metric_description`, `metric_category`, `metric_group`, `dashboard_name`)
 )
   ENGINE = InnoDB
   AUTO_INCREMENT = 0
 ;
+
+CREATE TABLE `stg_dict_business_metric` (
+  `metric_name` varchar(200) NOT NULL,
+  `metric_description` varchar(500) DEFAULT NULL,
+  `dashboard_name` varchar(100) DEFAULT NULL COMMENT 'Hierarchy Level 1',
+  `metric_group` varchar(100) DEFAULT NULL COMMENT 'Hierarchy Level 2',
+  `metric_category` varchar(100) DEFAULT NULL COMMENT 'Hierarchy Level 3',
+  `metric_sub_category` varchar(100) DEFAULT NULL COMMENT 'Additional Classification, such as Product, Line of Business',
+  `metric_level` varchar(50) DEFAULT NULL COMMENT 'CORE, DEPARTMENT, TEAM, OPERATION, STRATEGIC, TIER1, TIER2',
+  `metric_source_type` varchar(50) DEFAULT NULL COMMENT 'Table, Cube, File, Web Service',
+  `metric_source` varchar(300) CHARACTER SET latin1 DEFAULT NULL COMMENT 'Table Name, Cube Name, URL',
+  `metric_source_dataset_id` int(11) DEFAULT NULL COMMENT 'If metric_source can be matched in dict_dataset',
+  `metric_ref_id_type` varchar(50) CHARACTER SET latin1 DEFAULT NULL COMMENT 'DWH, ABTEST, FINANCE, SEGMENT, SALESAPP',
+  `metric_ref_id` varchar(100) CHARACTER SET latin1 DEFAULT NULL COMMENT 'ID in the reference system',
+  `metric_type` varchar(100) DEFAULT NULL COMMENT 'NUMBER, BOOLEAN, LIST',
+  `metric_additive_type` varchar(100) DEFAULT NULL COMMENT 'FULL, SEMI, NONE',
+  `metric_grain` varchar(100) DEFAULT NULL COMMENT 'DAILY, WEEKLY, UNIQUE, ROLLING 7D, ROLLING 30D',
+  `metric_display_factor` decimal(15,4) DEFAULT NULL,
+  `metric_display_factor_sym` varchar(20) DEFAULT NULL COMMENT '%, (K), (M), (B), (GB), (TB), (PB)',
+  `metric_good_direction` varchar(20) DEFAULT NULL COMMENT 'UP, DOWN, ZERO, FLAT',
+  `metric_formula` text COMMENT 'Expression, Code Snippet or Calculation Logic',
+  `dimensions` varchar(1000) CHARACTER SET latin1 DEFAULT NULL,
```

Comment:
Maybe we want the same char length for both stg and final table? So that when we load from stg to final, there won't be surprise issues.